src/components/Header/Header.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import {
+  FC, FormEvent, useCallback, useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+import { Errors } from '../../utils/enums';
+import { TodoData } from '../../types/TodoData';
+import { createTodo } from '../../api/todos';
+
+const USER_ID = 10284;
+
+interface Props {
+  todos: Todo[];
+  onAdd: (newTodo: Todo) => void;
+  setError: (error: null | Errors) => void;
+  setTempTodo: (newTodo: Todo | null) => void;
+  setIsLoading: (boolean: boolean) => void
+  loadTodos: () => void;
+}
+
+export const Header:FC<Props> = ({
+  todos,
+  onAdd,
+  setError,
+  setTempTodo,
+  setIsLoading,
+  loadTodos,
+}) => {
+  const [query, setQuery] = useState('');
+  const [isDisabled, setIsDisabled] = useState(false);
+
```

Comment:
I think the issue is in the Header component. Here is an example of suggested fix 
```
// Header Component
export const Header:FC<Props> = ({
  todos,
  onAdd,
  setError,
  setTempTodo,
  setIsLoading,
  loadTodos,
}) => {
  const [query, setQuery] = useState('');
  const [isDisabled, setIsDisabled] = useState(false);

  const handleSubmit = useCallback(async (event: FormEvent) => {
    event.preventDefault();

    if (!query) {
      setError(Errors.Title);

      return;
    }

    const todoData: TodoData = {
      completed: false,
      title: query,
      userId: USER_ID,
    };

    setTempTodo({ ...todoData, id: 0 });
    setIsLoading(true);
    try {
      setIsDisabled(true);
      const newTodo = await createTodo(todoData);

      onAdd(newTodo);
      setQuery('');
    } catch {
      setError(Errors.Add);
    } finally {
      // Load todos only after successfully adding a new one.
      // This prevents your tempTodo from disappearing.
      if (!setError) {
        loadTodos();
      }
      setTempTodo(null);
      setIsLoading(false);
      setIsDisabled(false);
    }
  }, [query]);

  // Remaining code...
};
```